### PR TITLE
Read Geotiff overviews into Pyramid struct

### PR DESCRIPTION
**Pull request by felixcremer**
This is a first approach and plotting is not working yet. But it is reading the overviews from a tiff.

### CODE REVIEW

**Comment by meggart** [comment]
Shouldn't we add some kind of abstraction here and define some kind of interface for different ways of storing and accessing pyramids? Initially we would have 2 implementations, one for [geozarr](https://github.com/zarr-developers/geozarr-spec/blob/main/geozarr-spec.md#multiscales) and the other one for everything that GDAL understands.

File: ext/PyramidSchemeArchGDALExt.jl [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module PyramidSchemeArchGDALExt
+using ArchGDAL: ArchGDAL as AG
+using PyramidScheme: PyramidScheme as PS
+using Rasters
+
+function PS.Pyramid(path::AbstractString)
```
[the remark this answers is not in the document]
Comment (felixcremer) [reply ext/PyramidSchemeArchGDALExt.jl:6]: Yes I think we can do it similar to how Rasters dispatches on the file extension to decide which backend to use. This is the file that defines the dispatch machinery and then in the extensions the corresponding method is defined:https://github.com/rafaqz/Rasters.jl/blob/main/src/sources/sources.jl

**Comment by felixcremer** [comment]
Yes we would need this abstraction and I am going to work on it when I work on the geozarr extension. The tests are currently falling, because we would need a new tag of ArchGDAL.

**Comment by felixcremer** [comment]
This is currently only working for two dimensional geotiffs, because we are explicitley only loading the first band.

**Comment by felixcremer** [comment]
This ended up to be more than adding geotiff overviews. This also adds broadcasting and fixes the plotting. I am going to merge this now, so that we can have a base to build on in the future.

**Pull request merged**